frontend/src/components/MinioArtifactLink.tsx

```diff
@@ -1,30 +1,36 @@
 import * as React from 'react';
-import { StoragePath, StorageService } from '../lib/WorkflowParser';
+import { StorageService } from '../lib/WorkflowParser';
 import { S3Artifact } from '../../third_party/argo-ui/argo_template';
-
-const artifactApiUri = ({ source, bucket, key }: StoragePath) =>
-  'artifacts/get' + `?source=${source}&bucket=${bucket}&key=${encodeURIComponent(key)}`;
+import { generateArtifactUrl } from '../lib/Utils';
+import { Apis } from '../lib/Apis';
 
 /**
- * A component that renders an artifact link.
+ * A component that renders a preview to an artifact with a link to the full content.
  */
 const MinioArtifactLink: React.FC<S3Artifact> = s3artifact => {
-  if (!s3artifact || !s3artifact.key || !s3artifact.bucket) {
+  const { key, bucket, endpoint } = s3artifact || {};
+  if (!key || !bucket) {
     return null;
   }
 
-  const { key, bucket, endpoint } = s3artifact;
+  const peek = 100;
+  const encodedKey = encodeURIComponent(key);
   const source = endpoint === 's3.amazonaws.com' ? StorageService.S3 : StorageService.MINIO;
-  const linkText = `${source.toString()}://${bucket}/${key}`;
+  const linkText = `${source.toString()}://${bucket}/${encodedKey}`;
+  const url = generateArtifactUrl(source, bucket, encodedKey);
+  const [content, setContent] = React.useState(linkText);
+
+  React.useEffect(() => {
```

Comment:
Congratulations! You are the first contributor to use react hooks in this repo. : )

Reply:
nit: you should return a clean up function to the hook. Without the cleanup, things like
1. first request out
2. props changed
3. second request out
4. second request back -> content set to be consistent with current props
5. first request back -> content set again consistent with outdated props
could happen. 

Although, we won't need to support changing props. We need the clean up function at least to avoid warnings like `React setState called when component is already unmount`

```
// an example simplified fix
useEffect(() => {
  let canceled = false;
  Apis.request().then(content => {
    if (!canceled) setContent(content);
  }
  return () => { canceled = false; };
}
```

Reply:
Btw I was unable to fix the unit test error.

Any ideas? I checked react and react Dom are of the right version. And everything builds fine.

I suspect it might be `enzyme-adapter-react-16`? But I am not really a react person.

```
MinioArtifactLink › handles s3 artifact
    Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
```

Reply:
Looks like we need to upgrade to latest enzyme: https://github.com/airbnb/enzyme/issues/2011

Reply:
Don't think it is going to be fixed soon. `useEffect` still not supported.
https://github.com/facebook/react/issues/17321

Reply:
I see, you can avoid testing it in shallow then. Just use mount.